net/device/ExceptionHelper.cs

```diff
@@ -0,0 +1,62 @@
+﻿// Microsoft.All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using DotNetty.Transport.Channels;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Net.Sockets;
+using System.Net.WebSockets;
+using System.Runtime.InteropServices;
+using System.Security.Authentication;
+
+namespace Microsoft.Azure.Iot.Thief.Device
+{
+    // Sample exception handler class - this class should be modified based on your application's logic
+    internal class ExceptionHelper
+    {
+        private static readonly HashSet<Type> s_networkExceptions = new HashSet<Type>
+        {
+            typeof(IOException),
+            typeof(SocketException),
+            typeof(ClosedChannelException),
+            typeof(TimeoutException),
+            typeof(OperationCanceledException),
+            typeof(HttpRequestException),
+            typeof(WebException),
+            typeof(WebSocketException),
```

Comment:
This list is interesting to me.  Elena wants to track how often different failures occur.  I wonder what granularity we want to support -- breaking down network exceptions into these 8 types could certainly be useful.  The tricky part is coming up with a way to be (relatively) consistent across languages with how we map failures into these groups.

Reply:
I copied it from Abhipsa's new sample https://github.com/Azure-Samples/azure-iot-samples-csharp/tree/master/iot-hub/Samples/device/DeviceReconnectionSample

Reply:
Just FYI - these are the exception types that the .NE sdk treats as transient network exceptions: https://github.com/Azure/azure-iot-sdk-csharp/blob/master/iothub/device/src/Transport/ErrorDelegatingHandler.cs#L29